### PR TITLE
Adding an example project.

### PR DESCRIPTION
**Pull request by finn-ball**
Adds an example project where we can point new users to get started with.

### CODE REVIEW

File: examples/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Copyright 2018 Google Inc.  All rights reserved
```
Comment (EdSchouten) [inline examples/WORKSPACE:1]: Where did these files come from? Did we modify them in any meaningful way?

If we didn't really modify them, couldn't we just point to the original location at which they are stored?

File: README.md [hunk 2]
```diff
@@ -72,3 +72,75 @@ the command below:
...
+```
+worker-ubuntu16-04_1  | xxxx/xx/xx xx:xx:xx rpc error: code = Unavailable desc = Failed to synchronize with scheduler: all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp xxx.xx.x.x:xxxx: connect: connection refused"
+```
+
+This is usually because container of the worker has started before the scheduler and so it cannot connect. After a second or so, this error message should stop.
+
+## Remote execution
+
+In a separate terminal, we will create a `.bazelrc` file and add a few extra options to it.
+```
```
Comment (EdSchouten) [inline README.md:99]: Does this part make some of the original documentation above superfluous? If so, be sure to remove parts of the old documentation.

File: WORKSPACE [hunk 3]
```diff
@@ -1 +1,50 @@
...
+load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig")
+
+rbe_autoconfig(name = "rbe_default")
+
+http_archive(
+    name = "abseil-hello",
+    urls = ["https://github.com/abseil/abseil-hello/archive/b4803b41ab3d58c503265148e5a7d3fd2a8e46d3.zip",],
+    strip_prefix = "abseil-hello-b4803b41ab3d58c503265148e5a7d3fd2a8e46d3/bazel-hello",
+    sha256 = "e676640e69e210636de795f571237bec09a9ad9af6e441bf56f0d193cfe1c9fc",
+)
```
Comment (EdSchouten) [inline WORKSPACE:50]: Could you run Buildifier on this file, please?

Note to self: we should have CI for this repository to explicitly fail when Buildifier/... isn't run.

File: README.md [hunk 4]
```diff
@@ -26,49 +26,76 @@ are established.
   <img src="https://github.com/buildbarn/bb-deployments/raw/master/bb-overview.png" alt="Overview of the Buildbarn setup"/>
 </p>
 
-## Using these deployments with Bazel
+# Getting started
 
-Bazel can perform remote builds against these deployments by adding
-[the official Bazel toolchain definitions](https://releases.bazel.build/bazel-toolchains.html)
-for the RBE container images to the `WORKSPACE` file of your project:
+This example aims to showcase a very simple build and test with remote execution using docker-compose as the deployment for Buildbarn. We will be compiling examples from the [abseill-hello](https://github.com/abseil/abseil-hello) project using Bazel.
```
Comment (EdSchouten) [inline README.md:31]: s/abseill/abseil/

File: kubernetes/storage.yaml [hunk 5]
```diff
@@ -17,7 +17,7 @@ spec:
       containers:
       - args:
         - /config/storage.jsonnet
-        image: buildbarn/bb-storage:20200505T185354Z-e1deeaf
+        image: buildbarn/bb-storage:20200320T182147Z-43e3483
```
Comment (EdSchouten) [inline kubernetes/storage.yaml:20]: This rolls back the container image versions?

File: docker-compose/docker-compose.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   frontend:
-    image: buildbarn/bb-storage:20200207T154538Z-f622cea
+    image: buildbarn/bb-storage:20200320T182147Z-43e3483
```
Comment (EdSchouten) [inline docker-compose/docker-compose.yml:4]: Be sure to upgrade to the latest container versions while you’re at it.

File: README.md [hunk 7]
```diff
@@ -26,49 +26,76 @@ are established.
...
+Bazel can perform remote builds against these deployments by adding [the official Bazel toolchain definitions](https://releases.bazel.build/bazel-toolchains.html) for the RBE container images to the `WORKSPACE` file of your project. It is also possible to derive your own configuration files using the `rbe_autoconfig`. More information can be found by reading the documentation [here](https://github.com/bazelbuild/bazel-toolchains/blob/master/rules/rbe_repo.bzl).
 
+For this example, we have provided the necessary `WORKSPACE` setup already but we still need to create the `.bazelrc`. In a separate terminal to your docker-compose deployment, create your `.bazelrc` in the project root:
 ```
-build:mycluster --bes_backend=grpc://fill-in-the-event-service-hostname-here:8985
-build:mycluster --bes_results_url=http://fill-in-the-browser-service-hostname-here/build_events/bb-event-service/
-build:mycluster --remote_executor=grpc://fill-in-the-frontend-service-hostname-here:8980
+cd bb-deployments/
+cp bazelrc .bazelrc
+echo "
```
Comment (EdSchouten) [inline README.md:57]: Maybe better to use:
```
cat >> .bazelrc << EOF
...
EOF
```
This is a bit more idiomatic for appending stuff to a text file.

File: docker-compose/config/worker-ubuntu16-04.jsonnet [hunk 8]
```diff
@@ -2,35 +2,36 @@ local common = import 'common.libsonnet';
 
 {
   blobstore: common.blobstore,
-  browserUrl: common.browserUrl,
   maximumMessageSizeBytes: common.maximumMessageSizeBytes,
   scheduler: { address: 'scheduler:8983' },
-  httpListenAddress: ':7986',
+  httpListenAddress: common.httpListenAddress,
```
Comment (EdSchouten) [inline docker-compose/config/worker-ubuntu16-04.jsonnet:7]: Is this change intentional? I think the Docker Compose based deployments don't try to renumber TCP ports. You also haven't updated docker-compose.yml to use the new port number?

Comment (finn-ball) [reply docker-compose/config/worker-ubuntu16-04.jsonnet:7]: It wasn't intentional. Reverting.
